src/web/services/auditTrailService.ts

```diff
@@ -0,0 +1,44 @@
+import axios from 'axios';
+
+import { AuditTrailDTO, AuditTrailEvents } from '../../api/entities/AuditTrail';
+import { mapClientTypeIdsToAdminEnums } from '../../api/services/adminServiceHelpers';
+import { backendError } from '../utils/apiError';
+import { getRoleNamesByIds } from '../utils/apiRoles';
+
+export const GetAuditLogs = async (participantId: number) => {
+  try {
+    const result = await axios.get<AuditTrailDTO[]>(`/participants/${participantId}/auditTrail`);
+    return result.data;
+  } catch (e: unknown) {
+    throw backendError(e, 'Could not get audit trail.');
+  }
+};
+
+export const getPrettyAuditDetails = (eventType: AuditTrailEvents, eventData: unknown) => {
+  const data = eventData as Record<string, unknown>;
+  const outputArray = [];
+  // eslint-disable-next-line guard-for-in
+  for (const key in data) {
+    let val = data[key];
+    // make data more human-readable
+    if (key.toLowerCase().indexOf('apiroles') > -1) {
+      const roleData = val as string[];
+      // some audit records use ids rather than names for the roles
+      if (roleData.length > 0 && !isNaN(Number(roleData[0]))) {
+        val = getRoleNamesByIds(val as number[]);
+      }
+    } else if (key.toLowerCase().indexOf('participanttypes') > -1) {
```

Comment:
Perhaps prettifying of each audit record type could be extracted to a separate method, to improve readability? it would be nice if it was something like:
```
for (const key in data) {
  let val = data[key];
  const lowerCaseKey = key.toLowerCase();
  if (lowerCaseKey.includes('apiRoles')) {
    val = prettifyApiRoleAuditDetails(val);
  } else if (lowerCaseKey.includes('participantTypes')) {
    val = prettifyParticipantTypeAuditDetails(val);
  }
```

Reply:
I mostly agree and that's what I started with, but it feels unsustainable as we continue to add different audit content and it runs counter to the way we create the audit logs (untyped data).  We can maybe revisit this but I think it does what we want it to do.